safe/report/processors/default.py

```diff
@@ -50,6 +50,32 @@
 LOGGER = logging.getLogger('InaSAFE')
 
 
+def composition_item(composer, item_id, item_class):
+    """Fetch a specific item according to its type in a composer.
+
+    We got some problem with QgsComposition::getComposerItemById. Don't use it,
+    and use this function instead.
+    See https://github.com/inasafe/inasafe/issues/4271
+
+    :param composer: The composer to look in.
+    :type composer: QgsComposition
+
+    :param item_id: The ID of the item to look for.
+    :type item_id: basestring
+
+    :param item_class: The expected class name.
+    :type item_class: cls
+
+    :return: The composition item, inherited class of QgsComposerItem.
+    """
+    for item in composer.items():
```

Comment:
before this line maybe we can check if the item class is QgsComposerMap and use `getComposerMapById` method instead to get the specific item that we want